themes/bootstrap3/js/common.js

```diff
@@ -94,14 +94,45 @@ var VuFind = (function VuFind() {
       }
     }
   };
-  var icon = function icon(name) {
+  var icon = function icon(name, attrs = {}) {
     if (typeof _icons[name] == "undefined") {
       console.error("JS icon missing: " + name);
       return name;
     }
 
     var html = _icons[name];
 
+    // Add additional attributes
+    function addAttrs(_html, _attrs = {}) {
+      var mod = String(_html);
+      for (var attr in _attrs) {
+        if (Object.prototype.hasOwnProperty.call(_attrs, attr)) {
+          var sliceStart = html.indexOf(" ");
+          var sliceEnd = sliceStart;
+          var value = _attrs[attr];
+          var regex = new RegExp(` ${attr}=(['"])([^\\1]+?)\\1`);
+          var existing = html.match(regex);
+          if (existing) {
+            sliceStart = existing.index;
+            sliceEnd = sliceStart + existing[0].length;
+            value = existing[2] + " " + value;
```

Comment:
Not sure what the use case is here, but wondering if we always want to concatenate attributes or if there might be situations where we're expecting to overwrite them. I'm thinking, for example, that below you have a call that sets custom class and title. It seems like you would want to append class, but overwrite title. Do we need to account for that somewhere? Maybe separate "addAttributes" and "overwriteAttributes" parameters?

Reply:
This function only exists inside the icon function. We can pull it out and expand it, but it was tailored to the specific cases the icon helper would need: adding to a minimal HTML tag.

Reply:
So your point is that we never expect this to be given HTML with anything other than a class attribute? If that's the case, I guess we don't need to go crazy overengineering it. :-)